src/kubelet/client/connector.go

```diff
@@ -228,6 +230,18 @@ func checkConnection(conn connParams) error {
 	return nil
 }
 
+func (dp *defaultConnector) pesterClientWithTimeout(timeout time.Duration) *pester.Client {
+	httpPester := pester.New()
+	httpPester.Backoff = pester.LinearBackoff
+	httpPester.MaxRetries = 3
+	httpPester.Timeout = timeout
+	httpPester.LogHook = func(e pester.ErrEntry) {
+		dp.logger.Debugf("getting data from kubelet: %v", e)
+	}
```

Comment:
Is this going to have issues because it is not able to connect to `kubelet` like with the agent?

Should we have this configurable in some way?

Reply:
I mean, if Kubelet has an issue and start rejecting this will backoff almost instantly as it happened to the agent.

Reply:
yes, but I would say that is linked to an other issue https://github.com/newrelic/nri-kubernetes/issues/322

I configured it in that way to mimic [KSM](https://github.com/newrelic/nri-kubernetes/blob/d67f7401273153b3dcc842f80323c7655d8a64f2/src/ksm/client/client.go#L39), but in the future I would make at least the timeout configurable per component